### ISSUE TITLE
Error reading loom file with read_loom()

### ISSUE THREAD

**Issue by yueqiw** [theislab/anndata#10]
Hi,

I converted a Seurat object into a `.loom` file and tried to read it into Scanpy using the `read_loom()` function. I got the following error: 

```
data = sc.read_loom(filename)

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-c4e72203067d> in <module>()
----> 1 data = sc.read_loom(filename)

/home/yueqi/anaconda/envs/py36/lib/python3.6/site-packages/anndata/readwrite/read.py in read_loom(filename)
    149         X.T,
    150         obs=lc.col_attrs,
--> 151         var=lc.row_attrs)
    152     lc.close()
    153     return adata

/home/yueqi/anaconda/envs/py36/lib/python3.6/site-packages/anndata/base.py in __init__(self, X, obs, var, uns, obsm, varm, raw, dtype, single_col, filename, filemode, asview, oidx, vidx)
    753                 obsm=obsm, varm=varm, raw=raw,
    754                 dtype=dtype, single_col=single_col,
--> 755                 filename=filename, filemode=filemode)
    756 
    757     def _init_as_view(self, adata_ref, oidx, vidx):

/home/yueqi/anaconda/envs/py36/lib/python3.6/site-packages/anndata/base.py in _init_as_actual(self, X, obs, var, uns, obsm, varm, raw, dtype, single_col, filename, filemode)
    889         # annotations
    890         self._obs = _gen_dataframe(obs, self._n_obs,
--> 891                                    ['obs_names', 'row_names', 'smp_names'])
    892         self._var = _gen_dataframe(var, self._n_vars, ['var_names', 'col_names'])
    893 

/home/yueqi/anaconda/envs/py36/lib/python3.6/site-packages/anndata/base.py in _gen_dataframe(anno, length, index_names)
    228                 break
    229         else:
--> 230             _anno = pd.DataFrame(anno)
    231     return _anno
    232 

/home/yueqi/anaconda/envs/py36/lib/python3.6/site-packages/pandas/core/frame.py in __init__(self, data, index, columns, dtype, copy)
    402                                          dtype=values.dtype, copy=False)
    403             else:
--> 404                 raise ValueError('DataFrame constructor not properly called!')
    405 
    406         NDFrame.__init__(self, mgr, fastpath=True)

ValueError: DataFrame constructor not properly called!
```

I revolved this error by changing `_anno = pd.DataFrame(anno)`
 to `_anno = pd.DataFrame(dict(anno))` in `base.py`. 

This is because the `loompy` package extracts column and row annotations as generators rather than dictionaries, and `pd.DataFrame` does not take generators as input. 

Hope it's fixed in the future. Thanks!

Yueqi

**Comment by falexwolf** [comment]
Thank you very much! This should do it: https://github.com/theislab/anndata/commit/d96a424377c0a78cd0b5b814b70b209d67dc3952

**Comment by falexwolf** [comment]
Since loompy 2.0, I'm also having trouble with this: https://github.com/theislab/anndata/blob/69abd62559f46eba5f6d93fc2a0a5d8860e32582/anndata/readwrite/write.py#L66 Does it work for you?

**Comment by yueqiw** [comment]
@falexwolf 

Thanks for the fix! 

I just tried `write_loom()` and got the following error at the same place as you mentioned:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-15-a95ba82ae600> in <module>()
----> 1 ad.readwrite.write_loom(filename="test1.loom", adata=data)

/home/yueqi/anaconda/envs/py36/lib/python3.6/site-packages/anndata/readwrite/write.py in write_loom(filename, adata)
     68         adata.X.T,
     69         row_attrs=row_attrs,
---> 70         col_attrs=col_attrs
     71     ) as ds:
     72         pass

AttributeError: __enter__

```

**Comment by falexwolf** [comment]
Ok, https://github.com/theislab/anndata/commit/b34609002537505b8b41b9a47099e9171b52b4b9 now fixes both read and write with `loompy>2.0`. Would be nice if you confirmed this. :wink:

Thank you for pointing me this!

**Comment by yueqiw** [comment]
@falexwolf Thanks! I tried read and write and they're both fixed.
